transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/BackendRegistry.java

```diff
@@ -33,6 +33,12 @@ void register(String name, TransportBackend backend) {
     }
   }
 
+  public Iterable<String> getAllBackendNames() {
+    synchronized (backends) {
+      return backends.keySet();
```

Comment:
This is not thread-safe, even though you have synchronization *within* the method, you're returning the underlying keySet, that can be modified after it is returned from this method.

That said, since we plan to remove `#register(String, TransportBackend)` it will not be a problem but for now please add a comment to document that this is not thread-safe and we need to fix it.